packages/shared/src/common/normalize-zip.ts

```diff
@@ -4,8 +4,17 @@
  * @returns The normalized zip code as a string.
  */
 export function normalizeZipCode(zipCode: string): string {
-  if (zipCode.includes("-") && zipCode.length === 9) {
-    return zipCode.slice(0, 4);
+  if (!isValidZipCode(zipCode)) {
+    throw new Error("Zip codes may only contain numbers (0-9) and a dash (-)");
   }
+  if (zipCode.includes("-") && zipCode.length === 9) return zipCode.slice(0, 4);
+  if (zipCode.length === 8) return zipCode.slice(0, 4);
   return zipCode.slice(0, 5);
 }
+
+function isValidZipCode(zipCode: string): boolean {
+  if (zipCode.length === 0) return false;
```

Comment:
nit: add `.trim()` before assessing length just in case?

Reply:
Good point